src/ISA/ISA/ArcTypes/CompositeHeader.fs

```diff
@@ -97,6 +101,34 @@ type CompositeHeader =
 
     with 
 
+    static member Cases = 
+        Microsoft.FSharp.Reflection.FSharpType.GetUnionCases(typeof<CompositeHeader>) 
+        |> Array.map (fun x -> x.Tag, x.Name)
+
+    /// <summary>
+    /// This function is used to programmatically create `CompositeHeaders` in JavaScript. Returns integer code representative of input type.
+    ///
+    /// 0: Expects no input
+    ///
+    /// 1: Expects OntologyAnnotation as input
+    ///
+    /// 2: Expects IOType as input
+    ///
+    /// 3: Expects string as input
+    /// </summary>
+    /// <param name="inp">Can be accessed from `CompositeHeader.Cases`</param>
+    static member jsGetColumnMetaType(inp:int) =
```

Comment:
Sry, I don't quite get this. Could you further describe this in the issue?

Reply:
programmatically decide how to create a specific CompositeHeader while looping over "Cases"

```js
it('jsGetColumnMetaType', function () {
        let cases = CompositeHeader.Cases
        let oa = OntologyAnnotation.fromString("My OA Name")
        let iotype = new IOType(0, [])
        let stringExample = "My Example"
        for (let mycase of cases) {
            let tag = mycase[0]
            let code = CompositeHeader.jsGetColumnMetaType(tag)
            switch (code) {
                case 0:
                    let header1 = new CompositeHeader(tag, [])
                    equal((header1.IsSingleColumn || header1.IsFeaturedColumn), true);
                    break;
                case 1:
                    let header2 = new CompositeHeader(tag, [oa])
                    equal(header2.IsTermColumn, true);
                    break;
                case 2:
                    let header3 = new CompositeHeader(tag, [iotype])
                    equal(header3.IsIOType, true);
                    break;
                case 3:
                    let header4 = new CompositeHeader(tag, [stringExample])
                    equal(header4.isFreeText, true);
                    break;
            }
        }
    })
```